lightwood/helpers/numeric.py

```diff
@@ -7,9 +7,10 @@
 
 Sorry, python is dumb, but the spirit of your investigation wasn't, try again
 '''
+from typing import Iterable
 
 
-def can_be_nan_numeric(value):
+def can_be_nan_numeric(value: object) -> bool:
```

Comment:
The logic of this function is not obvious at all; I recognize this is to explicitly handle np.nan-like situations where they can be converted into a str/float but not int representation, but we need to highlight this for others to understand. Please docstring the logic; it can be crude

Reply:
Will do